indexing-service/src/main/java/io/druid/indexing/overlord/ThreadPoolTaskRunner.java

```diff
@@ -335,6 +379,11 @@ public TaskStatus call()
 
       try {
         log.info("Running task: %s", task.getId());
+        TaskRunnerUtils.notifyLocationChanged(
```

Comment:
ok, I think it's fair to notify after the run is submitted. I can swap that.

But the real dirty secret is that nobody cares about what the ThreadPoolTaskRunner says a task's location is. Only the RTR and FTR matter. Actually the TPTR is kind of weird anyway- it supports all this cool thread pool stuff that nobody ever uses, since all use cases only submit a single task. Maybe TPTR shouldn't be a TaskRunner. It is kind of "special". any thoughts @drcrallen?